soc/nordic/ironside/include/nrf_ironside/uicr_def.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef IRONSIDE_INCLUDE_UICR_DEF_H__
+#define IRONSIDE_INCLUDE_UICR_DEF_H__
+
+#include <nrfx.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Common enum values */
+#define UICR_MAGIC_ERASE_VALUE (0xBD2328A8UL)         /* Default erased value for all UICR fields */
+#define UICR_DISABLED          UICR_MAGIC_ERASE_VALUE /* Common disabled value */
+#define UICR_ENABLED           (0xFFFFFFFFUL)         /* Common enabled value */
+#define UICR_PROTECTED         UICR_ENABLED           /* Common protected value */
+
+/**
+ * @brief APPROTECT [UICR_APPROTECT]
+ */
+struct UICR_APPROTECT {
+	__IOM uint32_t APPLICATION; /* APPLICATION access port protection */
+	__IOM uint32_t RADIOCORE;   /* RADIOCORE access port protection */
+
+	__IM uint32_t RESERVED;
+
+	__IOM uint32_t CORESIGHT; /* CoreSight access port protection */
```

Comment:
Seems a bit odd to use the `__IM` and `__IOM` definitions from the MDK.
How about replacing these with `volatile const` and `volatile`, respectively?

If we did that I think we could also remove the `#include <nrfx.h>` at the top.